File: level2/기능개발.py
import math
from collections import deque

def solution(progresses, speeds):
    answer = []
    
    q = deque()
    for i in range(len(progresses)):
        q.append(int(math.ceil((100-progresses[i])/speeds[i])))
        
    while q:
        n = 1
        x = q.popleft()
        
        while q and q[0] <= x:
            n += 1
            q.popleft()
        answer.append(n)
        
        
    
    return answer

File: level2/기능개발.cpp
#include <string>
#include <vector>
#include <queue>
#include <cmath>
using namespace std;

vector<int> solution(vector<int> progresses, vector<int> speeds) {
    vector<int> answer;
    queue<int> q;
    
    int size = progresses.size();
    for(int i = 0; i < size; i++){
        q.push((int)ceil((double)(100-progresses[i])/speeds[i]));
    }
    
    while(!q.empty()){
        int i = q.front(), n = 1;
        q.pop();
        while(!q.empty() && q.front() <= i){
            n++;
            q.pop();
        } 
        answer.push_back(n);
    }
    
    return answer;
}
